sync/purge-excess.js

```diff
@@ -39,39 +39,54 @@ function purgeExcess(dir, shrinkwrap, opts, cb) {
     deps.
 */
 function findExcess(dir, shrinkwrap, opts, cb) {
-    fs.readdir(dir, function (err, files) {
-        if (err) {
-            return cb(err);
-        }
-
-        files = files.map(function (file) {
-            return file.toLowerCase();
-        }).filter(function (file) {
-            // remove node_modules/.bin from check
-            return file !== '.bin';
-        });
-
-        if (opts.dev && shrinkwrap.devDependencies) {
-            var devDeps = shrinkwrap.devDependencies;
-            var devKeys = Object.keys(devDeps).map(function (s) {
-                return s.toLowerCase();
-            });
+    var files = [];
+    try {
+      files = fs.readdirSync(dir);
```

Comment:
because the `sync` command does fanout over the disk; it will be a lot slower synchronously.

Writing async logic is not hard; I generally ban sync logic.